standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/impl/ReadyToCleanAbortHandler.java

```diff
@@ -76,7 +76,13 @@ public class ReadyToCleanAbortHandler implements QueryHandler<List<CompactionInf
           " AND \"res1\".\"TC_TABLE\" = \"res3\".\"CQ_TABLE\" " +
           " AND (\"res1\".\"TC_PARTITION\" = \"res3\".\"CQ_PARTITION\" " +
           " OR (\"res1\".\"TC_PARTITION\" IS NULL AND \"res3\".\"CQ_PARTITION\" IS NULL))" +
-          " WHERE \"res3\".\"RETRY_RECORD_CHECK\" <= 0 OR \"res3\".\"RETRY_RECORD_CHECK\" IS NULL";
+          " WHERE (\"res3\".\"RETRY_RECORD_CHECK\" <= 0 OR \"res3\".\"RETRY_RECORD_CHECK\" IS NULL) " +
+              "AND NOT EXISTS (SELECT 1 " +
+              "                 FROM \"TXN_COMPONENTS\" AS \"txns\" " +
+              "                 WHERE \"txns\".\"TC_DATABASE\" = \"res1\".\"TC_DATABASE\" AND \"txns\".\"TC_TABLE\" = \"res1\".\"TC_TABLE\" " +
+              "                    AND (\"txns\".\"TC_PARTITION\" = \"res1\".\"TC_PARTITION\" " +
+              "                         OR (\"txns\".\"TC_PARTITION\" IS NULL AND \"res1\".\"TC_PARTITION\" IS NULL))" +
+              "                    AND \"txns\".\"TC_WRITEID\" > \"res1\".\"MAX_ABORTED_WRITE_ID\")";
```

Comment:
Why are we comparing records in TXN_COMPONENTS table? I thought the motive was to compare the records in COMPACTION_QUEUE to potentially avoid abort cleanup for tables which already have a COMPACTION_QUEUE record.

Reply:
Also this query lacks formatting.